src/components/ScmRepositoryUrl/ScmRepositoryUrl.tsx

```diff
@@ -32,23 +35,62 @@ const GerritButton = ({ isInline, url }: IGerritButtonProps) => (
   </TooltipWrapper>
 );
 
+/**
+ * The external button that redirect to specific external page.
+ *
+ * @param url - the external url for the SCM Repository
+ * @param isInline - whether to use inline style with external link action
+ */
+const ExternalUrlButton = ({ url, isInline }: IScmRepositoryUrlButtonProps) => {
+  if (!url) {
+    return <></>;
+  }
+  const urlResult = parseExternalRepositoryUrl({ externalUrl: url });
+  if (!urlResult.url) {
+    return <></>;
```

Comment:
it's somehow not possible to avoid a null result, because the result of the parse can be null. I slightly changed the code of`parseExternalRepositoryUrl` to return undefined instead of an object containing empty parameters when parse fails.